gammapy/utils/fitting/model.py

```diff
@@ -7,7 +7,11 @@
 class Model:
     """Model base class."""
 
-    # TODO: expose model parameters as attributes
+    __slots__ = []
```

Comment:
Have you tried to add `_parameters` to the `__slots__` here and remove it from all the sub-classes? I'm not sure whether it works. I guess right now we set the `self._parameters` in every sub-class, but this could happen as well in the base-class, when you pass the parameter list to `Model.__init__` anyway...

Reply:
Yes good point! I'll try that